concent_api/core/views.py

```diff
@@ -160,19 +199,34 @@ def handle_send_force_report_computed_task(client_message):
         client_message.TYPE,
         client_message.task_to_compute.compute_task_def['task_id'],
         client_message.serialize(),
-        status = ReceiveStatus
+        provider_public_key  = client_public_key,
+        requestor_public_key = other_party_public_key,
```

Comment:
You're never checking if `other_party_public_key` is not `None`. There is an `assert` inside `store_message_and_message_status()` but assertions are not for data validation. It won't protect you in production where assertion are disabled. The client can always send you a message without this header.

Reply:
base64.b64decode doesn't return None.

Reply:
But it does not come directly from `b64decode()`. It comes from `decode_other_party_public_key()`.

Reply:
I see that I misread the conditions in `decode_other_party_public_key()`. In that case it's OK.